api/src/services/cache/redis-cache.ts

```diff
@@ -6,20 +6,33 @@ import { CacheOptions, CacheService } from './cache';
 import { parse, stringify } from 'json-buffer';
 import { compress, decompress } from '../../utils/compress';
 import { map } from 'async';
+import { merge } from 'lodash';
 
 export class RedisCache extends CacheService {
 	client: RedisClientType;
 
 	constructor(options: CacheOptions) {
 		super(options);
 
-		this.client = createClient({
-			url: env.CACHE_REDIS || getConfigFromEnv('CACHE_REDIS_'),
-		});
+		this.client = createClient(this.getConfig());
 		this.client.on('error', (err) => logger.warn(err, `[cache] ${err}`));
 		this.client.connect();
 	}
 
+	private getConfig() {
+		if ('CACHE_REDIS' in env) return { url: env.CACHE_REDIS };
+		const config = merge({}, getConfigFromEnv('REDIS_'), getConfigFromEnv('CACHE_REDIS_'));
+		return {
+			socket: {
+				host: config.host,
+				port: config.port,
+			},
+			username: config?.username,
+			password: config?.password,
+			database: config?.db,
+		};
```

Comment:
Will the object with predefined properties returned in line 25 interfere or limit other use cases, such as TLS? Related issue in the past: #7724 (quite dated but should still be a valid use case for some users if I'm not mistaken)

Reply:
Good question!